frontend/app/services/user-info-service.ts

```diff
@@ -0,0 +1,29 @@
+import { z } from 'zod';
+
+export type UserInfo = {
+  firstName?: string;
+  lastName?: string;
+  phoneNumber?: string;
+};
+
+export async function getUserId() {
+  return '00000000-0000-0000-0000-000000000000';
+}
+
+export async function getUserInfo(userId: string) {
+  const response = await fetch(`https://api.example.com/users/${userId}`);
+  return z.custom<UserInfo>().parse(await response.json());
```

Comment:
https://github.com/colinhacks/zod?tab=readme-ov-file#custom-schemas
If you don't provide a validation function, Zod will allow any value. This can be dangerous!

>`z.custom<{` arg: string }>(); // performs no validation

Reply:
k.. I'll fix.